pyo3-arrow/src/table.rs

```diff
@@ -403,12 +403,14 @@ impl PyTable {
         &self,
         i: usize,
         field: NameOrField,
-        column: PyChunkedArray,
+        column: AnyArray,
     ) -> PyArrowResult<Arro3Table> {
+        let column = column.into_chunked_array()?;
         if self.num_rows() != column.len() {
-            return Err(
-                PyValueError::new_err("Number of rows in column does not match table.").into(),
-            );
+            return Err(PyValueError::new_err(
+                "The number of rows in column does not match the table.",
```

Comment:
It would probably be good of us to be more descriptive with these errors, to include in the error message how many rows were in each. We can make a new issue for that.